pkg/cli/cvd.go

```diff
@@ -278,18 +278,103 @@ func (c *cvdCreator) createCVDFromAndroidCI() ([]*hoapi.CVD, error) {
 }
 
 func (c *cvdCreator) createWithCanonicalConfig() ([]*hoapi.CVD, error) {
+	hostSrv := c.client.HostService(c.opts.Host)
+	envConfig := make(map[string]any)
+	deepCopy(c.opts.EnvConfig, &envConfig)
+	if err := c.uploadFilesFromEnvConfig(hostSrv, envConfig); err != nil {
+		return nil, fmt.Errorf("failed uploading files from environment config: %w", err)
+	}
 	createReq := &hoapi.CreateCVDRequest{
-		EnvConfig: c.opts.EnvConfig,
+		EnvConfig: envConfig,
 	}
 	c.statePrinter.Print(stateMsgFetchAndStart)
-	res, err := c.client.HostService(c.opts.Host).CreateCVD(createReq, c.credentialsFactory())
+	res, err := hostSrv.CreateCVD(createReq, c.credentialsFactory())
 	c.statePrinter.PrintDone(stateMsgFetchAndStart, err)
 	if err != nil {
 		return nil, err
 	}
 	return res.CVDs, nil
 }
 
+func (c *cvdCreator) uploadFilesFromEnvConfig(hostSrv hoclient.HostOrchestratorService, config map[string]interface{}) error {
```

Comment:
This function doesn't just upload the files, but also updates the config. Please rename to make this fact more evident.

Reply:
Good point. Done.